File: programmers/src/디스크컨트롤러.js
function solution(jobs) {
  let answer = 0
  let total = 0

  jobs.sort((a, b) => a[1] - b[1])
  console.log(jobs)
  for (const job of jobs) {
    total = total + total + job[1] - job[0]
    console.log(total)
  }

  answer = Math.floor(total / jobs.length)
  console.log(answer)
  return answer
}

const jobs = [
  [0, 3],
  [1, 9],
  [2, 6],
]
solution(jobs)

File: programmers/src/디스크컨트롤러.py
import heapq


def solution(jobs):
    answer, now, job_done = 0, 0, 0
    heap = []
    start = -1

    while job_done < len(jobs):
        for job in jobs:
            if start < job[0] <= now:
                heapq.heappush(heap, [job[1], job[0]])
        if heap:
            current = heapq.heappop(heap)
            start = now
            now += current[0]
            answer += now - current[1]
            job_done += 1
        else:
            now += 1
        print(now, start, heap)
    return int(answer / len(jobs))


solution([[0, 3], [1, 9], [2, 6]])
